2.0.0

BrooksResourceCentre opened the pull request:
Functional and well commented core version, ready for review before release.
Requires some additional documentation within the GitHub pages and possibly clean-up of samples but provides basic functionality

cnmhqwerty committed:
2.0.0 generalising interactions (#50)

* Implementation at an object level

#45

* Refactored input types and handling

#49

---------

Co-authored-by: Joshua Simpson <56636418+JoshuaOliver99@users.noreply.github.com>

skiletro committed:
update basic 360 video player to polished state (#51)

cnmhqwerty committed:
2.0.0 enforcing styling (#53)

* Create .editorconfig

#34 #33

* Auto re-formatted to style guide

skiletro committed:
change package.json version number

cnmhqwerty committed:
Grammar and version correction

UncleDracula commented on Samples~/360VideoPlayer/M_360VideoSphere.mat:
I'd personally try not to push metadata changes. I dont think anything here was an intentional change and likely was Unity auto. Changing to another one of our supported Unity versions will likely also make an edit.

cnmhqwerty replied:
We intend to standardise to 6000.2.2f1 for this reason, unintended change will try to catch this in future.

UncleDracula commented on Samples~/360VideoPlayer/VideoController.cs:
Style needs making consistent with other docs, e.g. [Tooltip][SerializeField]

cnmhqwerty replied:
[e03a105](https://github.com/BrooksResourceCentre/Unity-CAVE-SDK/pull/54/commits/e03a10576114786ba986a0dd674ab3a2bc467e97)

UncleDracula commented on Runtime/Input/InteractionObject.cs:
```diff
@@ -21,5 +21,8 @@ public abstract class InteractionObject : MonoBehaviour
         /// </example>
         public abstract void OnTouch();
+
+        public abstract void OnSwipe(Vector3 direction);
```
Should this be a vector3?

cnmhqwerty committed:
Update VideoController.cs

#54

UncleDracula commented on Runtime/Core/CAVEUtilities.cs:
```diff
@@ -1,52 +1,76 @@
-    /// <param name="pos"></param>
-    /// <param name="cameras"></param>
-    /// <returns></returns>
+    public enum TapTypes
```
Wondering if there's a more elegent solution than duplicating here?

I understand that these sets of abilities may be different

UncleDracula reviewed (changes requested):
To me, I feel that it could be clearer between:
CAVEMovement.cs
CAVEInputManager.cs
InputActionHandler.cs 

CAVEMovement.cs
- handles keyboard continuous movement 
- handles keyboard continuous rotation

CAVEInputManager.cs is doing only CAVE screen based actions?

InputActionHandler.cs 
- This is the touch actions only?
- Maybe we stop calling these actions to create a better distinction

UncleDracula commented:
Also, the package minimum unity version has been increased to 6?

cnmhqwerty committed:
Revert "Grammar and version correction"

This reverts commit b5990faf143aa3000167a8e66b38a65da954676a.

cnmhqwerty committed:
Update package.json

cnmhqwerty committed:
Removed Instantiate Projectile

#58 Problematic code was part of demo functionality so it has simply been removed as it's not required for core functionality

cnmhqwerty committed:
Removed duplicate enums

#57

skiletro committed:
#59 Add summaries and comment video controller script

skiletro committed:
Updated metadata files to reflect use of 2022.3.62f1

Under regular conditions, metafiles should not be committed. However,
these metafile changes are consistent for the Unity version supported,
and the version that a student or academic at the university may use.

cnmhqwerty committed:
Replaced swipe Vector3 with Vector2 direction

#55

cnmhqwerty committed:
Merge branch '2.0.0' of https://github.com/BrooksResourceCentre/Unity-CAVE-SDK into 2.0.0

cnmhqwerty committed:
Refactored names for core input classes in the package

#56

cnmhqwerty committed:
Update CAVE_Core.prefab

Updated default prefab settings to reference new script name

cnmhqwerty commented:
All previous comments resolved, package is now version agnostic just looking for your go ahead on these changes. @JoshuaOliver99
